### PR TITLE
[remix] Restore support for a custom `server.js` entrypoint file

### PR DESCRIPTION
**Pull request by TooTallNate**
When the project defines a custom `server.js` file in `remix.config.js`, then don't override that with the bundled
`server-node.mjs`/`server-edge.mjs` file. This allows for a custom `getLoadContext()` function to remain supported.

In order for Edge runtime to be supported, a [companion PR](https://github.com/remix-run/remix/pull/5537) for `@remix-run/vercel` adapter has been created with a dedicated entrypoint for when Edge runtime is being used. In order to support Edge on previous versions of the adapter, we may end up patching the package to enable support, but that will happen in a follow-up PR.

Related to https://github.com/orgs/vercel/discussions/1596.

### CODE REVIEW

File: packages/remix/src/build.ts [hunk 1]
```diff
@@ -339,17 +345,21 @@ async function createRenderNodeFunction(
...
+  let handler = relative(rootDir, serverBuildPath);
+  let handlerPath = join(rootDir, handler);
+  if (!serverEntryPoint) {
+    handler = join(dirname(handler), 'server-edge.mjs');
+    handlerPath = join(rootDir, handler);
 
-  // Copy the `server-edge.mjs` file into the "build" directory
-  const sourceHandlerPath = join(__dirname, '../server-edge.mjs');
-  await fs.copyFile(sourceHandlerPath, handlerPath);
+    // Copy the `server-node.mjs` file into the "build" directory
```
Comment (EndangeredMassa) [inline packages/remix/src/build.ts]: **suggestion (soft-blocking):** revert this line

```suggestion
    // Copy the `server-edge.mjs` file into the "build" directory
```

Was this changed in error? The code below this comment copies `server-edge.mjs`

File: packages/remix/test/fixtures/02-remix-basics-mjs/remix.config.mjs [hunk 2]
```diff
@@ -2,9 +2,13 @@
  * @type {import('@remix-run/dev').AppConfig}
  */
 export default {
-  ignoredRouteFiles: ['**/.*'],
+  ignoredRouteFiles: ["**/.*"],
+  // When running locally in development mode, we use the built-in remix
+  // server. This does not understand the vercel lambda module format,
+  // so we default back to the standard build output.
+  server: process.env.NODE_ENV === "development" ? undefined : "./server.ts",
+  //serverBuildPath: "api/index.js",
```
Comment (EndangeredMassa) [inline packages/remix/test/fixtures/02-remix-basics-mjs/remix.config.mjs]: **suggestion (non-blocking):** fix spacing

```suggestion
  // serverBuildPath: "api/index.js",
```

**Review by EndangeredMassa — DISMISSED** [review]

File: packages/remix/src/build.ts (hunk 1 above)
Comment (TooTallNate) [reply packages/remix/src/build.ts]: Ya good catch

**Review by EndangeredMassa — APPROVED** [review]

File: packages/remix/src/build.ts [hunk 3]
```diff
@@ -339,17 +345,21 @@ async function createRenderNodeFunction(
...
-  serverBuildPath: string
+  serverBuildPath: string,
+  serverEntryPoint: string | undefined
 ): Promise<EdgeFunction> {
   const files: Files = {};
 
-  const relativeServerBuildPath = relative(rootDir, serverBuildPath);
-  const handler = join(dirname(relativeServerBuildPath), 'server-edge.mjs');
-  const handlerPath = join(rootDir, handler);
+  let handler = relative(rootDir, serverBuildPath);
```
Comment (lauchness) [inline packages/remix/src/build.ts:353]: Does this change break support for Remix apps?

Comment (TooTallNate) [reply packages/remix/src/build.ts:353]: I don't understand the question, can you elaborate? This PR restores support for specifying an explicit `server.js` file in your Remix app.

**Pull request merged**